components/SearchBox.vue

```diff
@@ -0,0 +1,407 @@
+<template>
+  <div id="search" class="search-container" :class="{active: search_active}">
+    <div class="search-icon-container">
+      <div class="search-icon dark small">
+        <div class="mug-top" />
+        <div class="mug-sides" />
+        <div class="mug-bottom" />
+        <div class="mug-handle">
+          <div class="mug-handle-inner" />
+        </div>
+      </div>
+    </div>
+
+    <input
+      v-model="query"
+      type="text"
+      class="form-control search"
+      placeholder="Search beers, brewers, or styles"
+      aria-label="Search beers, brewers, or styles"
+      @focus="search_active = true"
+      @keyup.enter="onQueryEnter"
+    >
+    <button
+      class="btn btn-clearsearch btn-secondary"
+      type="button"
+      @click="onSearchClose"
+    >
+      <img src="/svg/close.svg" alt="close">
+    </button>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'SearchBox',
+  data () {
+    return {
+      query: undefined,
+      search_active: false
+    }
+  },
+  methods: {
+    onQueryEnter () {
+      this.$store.dispatch('beers/loadPage', {
```

Comment:
Not sure if we want to dispatch this here, or use a `this.$emit("myEvent")` and map that event to the parent. The parent component is a template, so.. feels weird.

Reply:
I don't know enough about Vue to have a strong opinion either way.